src/ScheduledTasks/Entity/Schedule.cs

```diff
@@ -0,0 +1,348 @@
+            this.logger.ResumedSchedule(this.State.ScheduleConfiguration.ScheduleId);
+
+            // compute next run based on startat and interval
+            context.SignalEntity(new EntityInstanceId(nameof(Schedule), this.State.ScheduleConfiguration.ScheduleId), nameof(this.RunSchedule), this.State.ExecutionToken);
+        }
+        catch (Exception ex)
+        {
+            this.logger.ScheduleOperationError(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, nameof(this.ResumeSchedule), "Failed to resume schedule", ex);
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Runs the schedule based on the defined configuration.
+    /// </summary>
+    /// <param name="context">The task entity context.</param>
+    /// <param name="executionToken">The execution token for the schedule.</param>
+    /// <exception cref="InvalidOperationException">Thrown when the schedule is not active or interval is not specified.</exception>
+    public void RunSchedule(TaskEntityContext context, string executionToken)
+    {
+        if (this.State.Status == ScheduleStatus.Uninitialized)
+        {
+            // this signal is no longer useful since the schedule has been deleted.
+            this.State = null!; // delete again, otherwise an uninitialized schedule will stick around
+            return;
+        }
+
+        ScheduleConfiguration scheduleConfig =
+            this.State.ScheduleConfiguration ??
+            throw new InvalidOperationException("Schedule configuration is missing.");
+        TimeSpan interval = scheduleConfig.Interval;
+
+        if (executionToken != this.State.ExecutionToken)
+        {
+            this.logger.ScheduleRunCancelled(scheduleConfig.ScheduleId, executionToken);
+            return;
+        }
+
+        if (this.State.Status != ScheduleStatus.Active)
+        {
+            string errorMessage = "Schedule must be in Active status to run.";
+            Exception exception = new InvalidOperationException(errorMessage);
+            this.logger.ScheduleOperationError(scheduleConfig.ScheduleId, nameof(this.RunSchedule), errorMessage);
+            throw exception;
+        }
+
+        // if endat is set and time now is past endat, do not run
+        if (scheduleConfig.EndAt.HasValue && DateTimeOffset.UtcNow > scheduleConfig.EndAt.Value)
+        {
+            this.logger.ScheduleRunCancelled(scheduleConfig.ScheduleId, executionToken);
+            this.State.NextRunAt = null;
+
+            context.SignalEntity(
+                new EntityInstanceId(nameof(Schedule), scheduleConfig.ScheduleId),
+                "delete",
+                this.State.ExecutionToken);
+
+            return;
+        }
+
+        this.State.NextRunAt = this.DetermineNextRunTime(scheduleConfig);
+
+        DateTimeOffset currentTime = DateTimeOffset.UtcNow;
+
+        if (this.State.NextRunAt!.Value <= currentTime)
+        {
+            this.StartOrchestration(context, currentTime);
```

Comment:
I was expecting that you would use `this.State.NextRunAt!.Value` instead of `now` on this line and the next line. So that the next runtime is based on the ideal time, not the actual time. Otherwise imprecisions add up over time. For example, if the user specifies start time 00:00:00 and interval 1h, we want the start time to be precisely on the hour for all subsequent iterations (not slowly diverging).